drake/solvers/OptimalMotionPlanningTree.m

```diff
@@ -0,0 +1,342 @@
+classdef OptimalMotionPlanningTree < TaskSpaceMotionPlanningTree
+  %Subclass of motion planning tree that override and add functions to be
+  %used with asymptotically optimal versions of RRT algorithms. It also
+  %adds the following properties
+%     C             cost array
+%     gamma         RRT* parameter
+%     steerFactor   radius for the steering function
+%     traj          indices of the optimal trajectory
+%     eta           radius for tree rewiring
+%     costType      type of cost
```

Comment:
it would be much more general (and not any harder?) to take in a function handle here, and perhaps provide length, etc as default options.